### PR TITLE
Implement liveness computation

### PR DESCRIPTION
**Pull request by albins**
This is a work in progress on issue #104.

Sorry for the poorly named branch; I started it to fix a silly compiler warning and ended up using it for the implementation.

### CODE REVIEW

**Comment by albins** [comment]
The current version seems to work, but I am not happy about generating an entire new `Relation` for walking the control-flow graph backwards.

I haven't yet written extensive tests, but I think it makes some sort of sense to go ahead with the extensions to rustc to produce the new facts so as to have some sort of ability to generate input data.

File: polonius-engine/src/output/naive.rs [hunk 1]
```diff
@@ -179,17 +203,17 @@ pub(super) fn compute<Region: Atom, Loan: Atom, Point: Atom, Variable: Atom>(
...
+                    .or_insert_with(BTreeSet::new)
                     .insert(*borrow);
             }
 
             for (region, location) in &region_live_at_rel.elements {
                 result
                     .region_live_at
                     .entry(*location)
-                    .or_insert(vec![])
+                    .or_insert_with(|| vec![])
```
Comment (shepmaster) [inline polonius-engine/src/output/naive.rs]: This should be `(Vec::new)` as below.

Comment (albins) [reply polonius-engine/src/output/naive.rs]: Thanks! Fixed!

File: src/intern.rs [hunk 2]
```diff
@@ -24,6 +24,10 @@ where
         &self.rev_strings[data]
     }
 
+    pub(crate) fn untern_vec(&self, data: &Vec<TargetType>) -> Vec<&str> {
```
Comment (shepmaster) [inline src/intern.rs]: https://stackoverflow.com/q/40006219/155423

Comment (albins) [reply src/intern.rs]: Thanks! I really appreciate these comments!

Changed data to be a slice of `TargetType`.

**Comment by albins** [comment]
This is marked WIP again as I might as well roll the entire region logic into this PR as per #104.

**Pull request merged**